runtime/include/comm/ugni/chpl-comm-impl.h

```diff
@@ -21,6 +21,28 @@
 #define _chpl_comm_impl_h_
 
 //
+// This is the comm layer sub-interface for dynamic allocation and
+// registration of (typically) array memory.
```

Comment:
I'd drop the `(typically) array` since this will (hopefully) change in the future